docs/rest.md

```diff
@@ -0,0 +1,623 @@
+      (e.g. installing), the operation field describes that operation
+    * `rollback_available`: if present and not empty, it means the package can
+      be rolled back to the version specified as a value to this entry.
+    * `update_available`: if present and not empty, it means the package can be
+      updated to the version specified as a value to this entry.
+
+### POST
+
+* Description: Sideload a package to the system. 
+* Authorization: trusted
+* Operation: async
+* Return: background operation or standard error
+
+#### Input
+
+The package to sideload should be provided as part of the body of a
+`mutlipart/form-data` request. The form should have only one file. If it also
+has an `allow-unsigned` field (with any value), the package may be unsigned;
+otherwise attempting to sideload an unsigned package will result in a failed
+background operation.
+
+It's also possible to provide the package as the entire body of a `POST` (not a
+multipart request). In this case the header `X-Allow-Unsigned` may be used to
+allow sideloading unsigned packages.
+
+### PUT
+
+* Description: change configuration for active packages. It is an error to
+  attempt to change configuration for non-active packages; if a configuration
+  change is requested for a package that is not active in the system the whole
+  command is aborted even if other packages that are active are specified in
+  the same command.
+* Authorization: trusted
+* Operation: sync
+* Return: configuration for all listed packages
+
+The request body is expected to be a JSON object with keys being the qualified
+package name(s) to configure, and the values will be passed to the configure
+hooks of the packages. The background operation information will similarly
+list individual statuses of the configuration changes.
+
+#### Sample input:
+
+```javascript
+{
+   "dd.canonical": "some-option: 42",
+   "hello-world.canonical": "greeting: Hi"
+}
+```
+
+#### Sample result:
+
+```javascript
+{
+   "dd.canonical": "some-option: 42\nsome-other-option: true",
+   "hello-world.canonical": "greeting: Hi\nheader: false"
+}
+```
+
+## /1.0/packages/[name]
+### GET
+
+* Description: Details for a  package
+* Authorization: trusted
+* Operation: sync
+* Return: package details (as in `/1.0/packages/`)
+
+
+### POST
+
+* Description: Install, update, remove, purge, activate, deactivate, or
+  rollback the package
+* Authorization: trusted
+* Operation: async
+* Return: background operation or standard error
+
+#### Sample input
+
+```javascript
+{
+ "action": "install"
+}
+```
+
+#### Fields in the input object
+
+field      | ignored except in action | description
+-----------|-------------------|------------
+`action`   |                   | Required; a string, one of `install`, `update`, `remove`, `purge`, `activate`, `deactivate`, or `rollback`.
+`leave_old`| `install` `update` `remove` | A boolean, default is false (do not leave old packages around). Equivalent to commandline's `--no-gc`.
```

Comment:
I'm not sure what the problem is, but I'm pretty sure adding "garbage collect" in this context will not help :-)

Without the option, or explicitly setting it to `false`, it will remove old packages. With the option, it will leave them.